test-project/Assets/PlaymodeTests/SubscriptionAggregateCouplingTests.cs

```diff
@@ -0,0 +1,171 @@
+        [TestCase(typeof(TestMonoBehaviours.Entity))]
+        [TestCase(typeof(TestMonoBehaviours.LinkedGameObjectMap))]
+        [TestCase(typeof(TestMonoBehaviours.LogDispatcher))]
+        [TestCase(typeof(TestMonoBehaviours.WorkerFlagReader))]
+        [TestCase(typeof(TestMonoBehaviours.WorkerId))]
+        [TestCase(typeof(TestMonoBehaviours.WorldCommandSender))]
+        [TestCase(typeof(TestMonoBehaviours.World))]
+        public void DifferentAggregateSubscriptions_should_not_couple_together(Type requireableType)
+        {
+            var method = typeof(SubscriptionAggregateCouplingTests)
+                .GetMethod("DifferentAggregateSubscriptions_should_not_couple_together_impl",
+                    BindingFlags.Instance | BindingFlags.NonPublic);
+
+            Assert.NotNull(method);
+
+            method.MakeGenericMethod(requireableType).Invoke(this, new object[] { });
+        }
+
+        // This tests a bug we encountered where two different GameObjects that both had subscriptions that
+        // were cached incorrectly would cause the subscriptions to become coupled.
+        private void DifferentAggregateSubscriptions_should_not_couple_together_impl<T>() where T : MonoBehaviour
+        {
+            var gameObject1 = CreateAndLinkGameObject<T>(1);
+            var gameObject2 = CreateAndLinkGameObject<T>(2);
+
+            RemoveAndUnlinkGameObject(1, gameObject1);
+            RemoveAndUnlinkGameObject(2, gameObject2);
+
+            var gameObject3 = CreateAndLinkGameObject<T>(3);
+            var gameObject4 = CreateAndLinkGameObject<T>(4);
+
+            RemoveAndUnlinkGameObject(3, gameObject3);
+
+            Assert.IsTrue(gameObject4.GetComponent<T>().enabled, "gameObject4.GetComponent<T>().enabled");
+        }
+
+        private GameObject CreateAndLinkGameObject<T>(long entityId) where T : MonoBehaviour
+        {
+            var template = new EntityTemplate();
+            template.AddComponent(new Position.Snapshot(), "worker");
+            connectionHandler.CreateEntity(entityId, template);
+            receiveSystem.Update();
+
+            var gameObject = new GameObject("TestGameObject");
+            gameObject.AddComponent<T>();
+            gameObject.GetComponent<T>().enabled = false;
+
+            linker.LinkGameObjectToSpatialOSEntity(new EntityId(entityId), gameObject);
+            requireLifecycleSystem.Update();
+
+            return gameObject;
+        }
+
+        private void RemoveAndUnlinkGameObject(long entityId, GameObject gameObject)
+        {
+            connectionHandler.RemoveComponent(entityId, Position.ComponentId);
+            connectionHandler.RemoveEntity(entityId);
+            receiveSystem.Update();
+            requireLifecycleSystem.Update();
+
+            Object.DestroyImmediate(gameObject);
+        }
+
+        // Drop all logs, we don't care about them for this test.
+        private class IgnoreLogsDispatcher : ILogDispatcher
```

Comment:
Is this still something that's true? It feels easy to erase errors with this.

Reply:
I can check to see if it fires off any log messages now. Sadly we'd need to "expect" all of them which is slightly annoying 😕

Reply:
Didn't need this in the end 💯